### PR TITLE
Update project

### PR DESCRIPTION
**Pull request by AVK1206**
list task review

### CODE REVIEW

File: templates/todo/tag_list.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                <tr>
+                    <td>{{ tag.name }}</td>
+                    <td>
+                        <button style="background: gray" class="btn"><a role="button" style="color: white"
+                                                                        href="{% url 'todo:task_change_status' pk=tag.id %}">Update</a>
+                        </button>
+                    </td>
+                    <td>
+                        <button style="background: gray" class="btn"><a role="button" style="color: white"
+                                                                        href="{% url 'todo:task_change_status' pk=tag.id %}">Delete</a>
```
Comment (AVK1206) [inline templates/todo/tag_list.html:27]: Both Update and Delete buttons link to the same URL — likely an error. Verify and update the Delete link to point to the correct URL

File: templates/todo/task_confirm_delete.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+{% block content %}
+
+    <h1>Delete task</h1>
+
+    <p>Are you sure you want to delete the task {{ task }}</p>
+
+    <form action="" method="post">
+    {% csrf_token %}
+        <input type="submit" value="Yes, delete" class="btn btn-danger">
+    </form>
```
Comment (AVK1206) [inline templates/todo/task_confirm_delete.html:12]: Consider adding a `cancel` button to let users back out easily instead of only having the delete confirmation

File: templates/todo/task_list.html
Comment (AVK1206) [inline templates/todo/task_list.html:1]: Consider improving accessibility by adding `aria-label` and  moving CSS styles into the stylesheet as in comment above

File: todo/forms.py
Comment (AVK1206) [inline todo/forms.py:1]: Consider adding validation methods if you want to enforce custom constraints on form fields

File: todo/models.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    class Meta:
+        ordering = ["name"]
+
+    def __str__(self):
+        return f"{self.name}"
+
+
+class Task(models.Model):
+    content = models.CharField(blank=True, max_length=15)
```
Comment (AVK1206) [inline todo/models.py:15]: `max_length=15`  feels very short for a task description. Consider increasing the length

File: todo/models.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    created_at = models.DateTimeField(auto_now_add=True)
+    deadline = models.DateTimeField()
```
Comment (AVK1206) [inline todo/models.py:17]: Consider adding validation to ensure deadline is not set in the past

File: todo/tests.py
Comment (AVK1206) [inline todo/tests.py:1]: Add relevant test cases to cover key functionalities and corner cases to ensure code reliability and maintainability

File: todo/urls.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    TaskListView,
+    TaskCreateView,
+    TaskUpdateView,
+    TaskDeleteView,
+    TagListView,
+    TagCreateView,
+    TagUpdateView,
+    TagDeleteView,
+    TaskChangeStatusView,
+
```
Comment (AVK1206) [inline todo/urls.py:13]: Remove extra blank line

File: todo/urls.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    path("task/create", TaskCreateView.as_view(), name="task-create"),
+    path("task/<int:pk>/update", TaskUpdateView.as_view(), name="task-update"),
+    path("task/<int:pk>/delete", TaskDeleteView.as_view(), name="task-delete"),
+
+    path("tags/", TagListView.as_view(), name="tag-list"),
+    path("tags/create", TagCreateView.as_view(), name="tag-create"),
+    path("tags/<int:pk>/update", TagUpdateView.as_view(), name="tag-update"),
+    path("tags/<int:pk>/delete", TagDeleteView.as_view(), name="tag-delete"),
+
+    path('chng_st/<int:pk>', TaskChangeStatusView.as_view(), name='task_change_status'),
```
Comment (AVK1206) [inline todo/urls.py:27]: Consider renaming the path from `chng_st` to something more descriptive like `change_status` for better readability

File: todo/urls.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
```
Comment (AVK1206) [inline todo/urls.py:28]: and remove here extra blank line

File: todo/views.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+class TagDeleteView(generic.DeleteView):
+    model = Tag
+    fields = "__all__"
+    template_name = "todo/tag_confirm_delete.html"
+    success_url = reverse_lazy("todo:tag-list")
+
+
+class TaskChangeStatusView(View):
+    def post(self, request, pk):
```
Comment (AVK1206) [inline todo/views.py:67]: Consider adding type hints

File: todo/views.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        task = Task.objects.get(id=pk)
+        task.is_done = not task.is_done
+        task.save()
+
+        return redirect(reverse_lazy('todo:task-list'), permanent=True)
+
+    def get(self, request):
```
Comment (AVK1206) [inline todo/views.py:74]: type hints as well

File: todo/views.py [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    template_name = "todo/task_list.html"
+    paginate_by = 3
+    queryset = Task.objects.prefetch_related("tags")
+
+
+class TaskCreateView(generic.CreateView):
+    model = Task
+    form_class = TaskForm
+    template_name = "todo/task_form.html"
+    success_url = "http://127.0.0.1:8000/task/create"
```
Comment (AVK1206) [inline todo/views.py:21]: Avoid hardcoding URLs and use `reverse_lazy("todo:task-create")`. This improves portability and environment flexibility

File: todo_list/settings.py [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+from pathlib import Path
+
+# Build paths inside the project like this: BASE_DIR / 'subdir'.
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/4.0/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = "django-insecure-r#4o00zb3g*%diq!@(qe7n*u4(u6z!$z@3cm7ch*npwr=@qr7s"
```
Comment (AVK1206) [inline todo_list/settings.py:19]: Avoid committing the actual SECRET_KEY to version control. Use an environment variable from file instead and don't forget to add that file to .gitignore and .dockerignore so on. No one should see `SECRET_KEY` it's a real security issue
